Pegasus/include/particleforcegenerator.hpp

```diff
@@ -1,147 +1,181 @@
-#ifndef PEGAS_PARTICLE_FORCE_GENERATOR_HPP
-#define PEGAS_PARTICLE_FORCE_GENERATOR_HPP
+#ifndef PEGASUS_PARTICLE_FORCE_GENERATOR_HPP
+#define PEGASUS_PARTICLE_FORCE_GENERATOR_HPP
 
 #include "Pegasus/include/particle.hpp"
 #include <map>
 #include <memory>
 #include <set>
 #include <vector>
 
-namespace pegas {
+namespace pegasus {
+
 class ParticleForceGenerator {
 public:
-    using Ptr = std::shared_ptr<ParticleForceGenerator>;
-    using ConstPtr = std::shared_ptr<ParticleForceGenerator const>;
-
     virtual ~ParticleForceGenerator() {}
-
-    virtual void updateForce(Particle::Ptr const& p) = 0;
+    virtual void updateForce(Particle & p) = 0;
 };
 
 class ParticleForceRegistry {
 public:
-    using Ptr = std::shared_ptr<ParticleForceRegistry>;
-
-    void add(Particle::Ptr& p, ParticleForceGenerator::Ptr& pfg);
-
-    void remove(Particle::Ptr const& p);
-
-    void remove(Particle::Ptr const& p, ParticleForceGenerator::Ptr const& pfg);
-
+    void add(Particle & p, ParticleForceGenerator & pfg);
+    void remove(Particle & p);
+    void remove(Particle & p, ParticleForceGenerator & pfg);
     void clear();
-
     void updateForces();
 
 private:
-    using Registry = std::map<Particle::Ptr, std::set<ParticleForceGenerator::Ptr> >;
-    Registry mRegistrations;
+    std::map<Particle*, std::set<ParticleForceGenerator*>> mRegistrations;
```

Comment:
`map` also stores the items ordered which may be suboptimal for pointer keys. Could you write some test cases for this particular use to see if there is any noticeable difference?